src/application-container/private/ApplicationContainerErrorBoundary.jsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import StatusView from 'terra-status-view';
+
+import ApplicationIntlContext from '../../application-intl/ApplicationIntlContext';
+import ErrorBoundaryBase from '../../shared/ErrorBoundaryBase';
+
+const propTypes = {
+  /**
+   * Components to render within the context of the error boundary. Exceptions
+   * thrown by these components during their render lifecycle will be caught by
+   * the error boundary and will result in an error view being rendered instead.
+   */
+  children: PropTypes.node,
+};
+
+/**
+ * The ApplicationContainerErrorBoundary provides a render error boundary for
+ * an entire application.
+ *
+ * It should not be used to provide error handling within page content.
+ */
+const ApplicationContainerErrorBoundary = ({ children }) => {
+  const applicationIntl = React.useContext(ApplicationIntlContext);
+  const [caughtError, setCaughtError] = React.useState(false);
```

Comment:
i'm curious, any particular reason we chain the context/state calls from the React object, instead of importing them separately first?

Reply:
There were issues in the past with mocking these individual module properties due to how module scoping works with Jest. Depending on how you test, it's not a big deal, but I got in the habit of doing it this way just in case.

Reply:
that's good to know!